src/librbd/Journal.cc

```diff
@@ -1349,6 +1350,10 @@ void Journal<I>::handle_replay_process_safe(ReplayEntry replay_entry, int r) {
   CephContext *cct = m_image_ctx.cct;
 
   m_lock.Lock();
+  assert(m_state == STATE_REPLAYING ||
+         m_state == STATE_FLUSHING_RESTART ||
+         m_state == STATE_FLUSHING_REPLAY);
```

Comment:
@dillaman Observing this assert violation in TestJournalReplay.Rename:
http://qa-proxy.ceph.com/teuthology/trociny-2016-06-07_23:26:11-rbd-wip-mgolub-testing---basic-vps/245308/teuthology.log